internal/services/appconfiguration/app_configuration_resource.go

```diff
@@ -64,6 +99,17 @@ func resourceAppConfiguration() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"soft_delete_retention_days": {
+				Type:         pluginsdk.TypeInt,
+				Optional:     true,
+				Default:      7,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(1, 7),
+				DiffSuppressFunc: func(_, old, new string, _ *pluginsdk.ResourceData) bool {
+					return old == "0"
```

Comment:
Here this because service default value is 0 for free sku,  but 7 for standard sku.